user/build-stages.md

```diff
@@ -0,0 +1,186 @@
+
+This screencast demonstrates how the two stages work:
+
+![](https://cloud.githubusercontent.com/assets/3729517/25229553/0868909c-25d1-11e7-9263-b076fdef9288.gif)
+
+## Naming your stages
+
+Stage names are arbitrary strings, and can include spaces or emojis. The first
+letter of a stage name is automatically capitalized for aesthetical reasons, so
+you don't have to deal with uppercase strings in your `.travis.yml` file.
+
+The default stage is `test`. Jobs that do not have a stage name are assigned to
+the previous stage name if one exists, or the default stage name if there is no
+previous stage name. `test`. This means that if you set the stage name on the
+first job of each stage, the build will work as expected.
+
+For example the following config is equivalent to the one above, but also adds a
+second deploy job to the `deploy` stage that deploys to a different target. As
+you can see you only need to specify the stage name once:
+
+```yaml
+jobs:
+  include:
+    - script: ./test 1 # uses the default stage name "test"
+    - script: ./test 2
+    - stage: deploy
+      script: ./deploy target-1
+    - script: ./deploy target-2
+```
+
+## Build stages and build matrix expansion
+
+[Matrix expansion](https://docs.travis-ci.com/user/customizing-the-build/#Build-Matrix)
+means that certain top level configuration keys expand into a matrix of jobs.
+
+For example:
+
+```yaml
+env:
+  - FOO=foo
+  - FOO=bar
+jobs:
+  include:
+    - stage: deploy
+      env:
+        - FOO=foo
+      script: ./deploy
+```
+
+This will run two jobs with the env vars `FOO=foo` and `FOO=bar` respectively
+first, and assign these to the default stage test. The third job on the deploy
+stage starts only after the test stage has completed successfully. Be sure to
+set the env var `FOO` if your script takes it into account.
+
+TODO: might make more sense to skip the first example and just use the rvm one? I think it is clearer
+
+For example if you use `rvm` (or any other language runtime key) to specify
+runtime versions, you **must** also specify the `rvm` version on the included
+deploy job:
+
+
+```yaml
+rvm:
+  - 2.3
+  - 2.4
+jobs:
+  include:
+    - stage: deploy
+      rvm: 2.4
+      env:
+        - FOO=foo
+      script: ./deploy
+```
+
+## Build stages and deployments
+
+You can combine build stages with our [deployment integration](https://docs.travis-ci.com/user/deployment/):
+
+```yaml
+jobs:
+  include:
+    - script: ./test 1 # uses the default stage name "test"
+    - script: ./test 2
+    - stage: deploy
+      script: skip     # usually you do not want to rerun any tests
+      deploy: &heroku
+        provider: heroku
+        # ...
+```
+
+Travis CI does not set or overwrite any of your scripts, and most languages
+have a [default test script](https://docs.travis-ci.com/user/languages/ruby/#Default-Test-Script)
+defined. So in many use cases you might want to overwrite the `script` by
+specifying the keyword `skip` or `ignore`.
+
+## Examples
+
```

Comment:
I cut the mini menu, because you're just linking to the heading, and then again to the page, so it feels like an extra jump.